BDArmory/Weapons/Missiles/MissileLauncher.cs

```diff
@@ -1591,19 +1598,19 @@ IEnumerator FireReloadableMissile()
             ml.guidanceActive = true;
 
             BDATargetManager.FiredMissiles.Add(ml);
-            if (wpm != null)
+            if (FiredByWM != null)
             {
-                ml.Team = wpm.Team;
-                wpm.SendTargetDataToMissile(ml, targetVessel != null ? targetVessel.Vessel : null, true, new MissileFire.TargetData(targetGPSCoords, TimeOfLastINS, INStimetogo), true);
-                wpm.heatTarget = TargetSignatureData.noTarget;
-                ml.GpsUpdateMax = wpm.GpsUpdateMax;
-                wpm.UpdateQueuedLaunches(targetVessel, ml, false);
-                wpm.UpdateMissilesAway(targetVessel, ml);
+                ml.Team = FiredByWM.Team;
+                FiredByWM.SendTargetDataToMissile(ml, targetVessel != null ? targetVessel.Vessel : null, true, new MissileFire.TargetData(targetGPSCoords, TimeOfLastINS, INStimetogo), true);
+                FiredByWM.heatTarget = TargetSignatureData.noTarget;
+                ml.GpsUpdateMax = FiredByWM.GpsUpdateMax;
+                FiredByWM.UpdateQueuedLaunches(targetVessel, ml, false);
+                FiredByWM.UpdateMissilesAway(targetVessel, ml);
             }
 
             if (ml.radarTarget.exists && ml.radarTarget.lockedByRadar && ml.radarTarget.lockedByRadar.vessel != ml.SourceVessel)
             {
-                MissileFire datalinkwpm = ml.radarTarget.lockedByRadar.vessel.ActiveController().WM;
+                MissileFire datalinkwpm = ml.radarTarget.lockedByRadar.vessel.ActiveController().WM; // FIXME SI Does this matter if the primary WM on the datalink changes?
                 if (datalinkwpm)
```

Comment:
If keeping the 'WM doesn't match, delete missile' stuff on 1475, no, because the code wouldn't execute to this point.
Otherwise, in the case that the parent ship has a radar and the child ship doesn't, it wouldn't be any different from launching a missile using a second ship's radar via a Radar Data Receiver part, and the missile's datalink WM would be set to the parent's WM for linking to the locking radar set, so this should be correct as is.